beacon-chain/db/slasherkv/slasher.go

```diff
@@ -307,6 +310,34 @@ func (s *Store) CheckDoubleBlockProposals(
 	return proposerSlashings, err
 }
 
+// BlockProposalForValidator given a validator index and a slot
+// retrieves an existing proposal record we have stored in the database.
+func (s *Store) BlockProposalForValidator(
```

Comment:
This function is not used outside tests. Is it possible to move it to the tests package and exclude from the interface?

Reply:
The problem is we need some method to retrieve this from the DB, and it needs to be part of the DB interface methods. It will be important as well in the future for API endpoints, so I think it is safe to include